keps/sig-cluster-lifecycle/kubeadm/4214-separate-super-user-kubeconfig/README.md

```diff
@@ -0,0 +1,724 @@
+-->
+
+On upgrade, the regular kubeconfig and certificate renewal process will be
+performed. The `admin.conf` file will be replaced with a file that has
+the de-escalated privileges. If the `super-admin.conf` file is present on
+a node during a future N+1 release, the file will be replaced with
+updated credentials. If `super-admin.conf` is not present, no errors
+will be returned.
+
+The `kubeadm upgrade apply` command will manage the addition of
+the new ClusterRoleBinding `kubeadm:cluster-admins`. One release
+after the feature was enabled, this logic for the RBAC management
+can be removed.
+
+kubeadm does not support downgrades.
+
+### Version Skew Strategy
+
+<!--
+If applicable, how will the component handle version skew with other
+components? What are the guarantees? Make sure this is in the test plan.
+
+Consider the following in developing a version skew strategy for this
+enhancement:
+- Does this enhancement involve coordinating behavior in the control plane and nodes?
+- How does an n-3 kubelet or kube-proxy without this feature available behave when this feature is used?
+- How does an n-1 kube-controller-manager or kube-scheduler without this feature available behave when this feature is used?
+- Will any other components on the node change? For example, changes to CSI,
+  CRI or CNI may require updating that component before the kubelet.
+-->
+
+The target release of kubeadm will be able to upgrade from N-1 nodes that
+do not have the feature enabled yet. During cluster creation with the
+target kubeadm version the feature will become enabled immediately.
+
+The `kubeadm upgrade apply` command will manage the addition of
+the new ClusterRoleBinding `kubeadm:cluster-admins` RBAC. One release
+after the feature was enabled, this logic for the RBAC management
+can be removed.
+
+## Production Readiness Review Questionnaire
+
+Not applicable for kubeadm. The kubeadm project is considered "out-of-tree".
+
+## Implementation History
+
+<!--
+Major milestones in the lifecycle of a KEP should be tracked in this section.
+Major milestones might include:
+- the `Summary` and `Motivation` sections being merged, signaling SIG acceptance
+- the `Proposal` section being merged, signaling agreement on a proposed design
+- the date implementation started
+- the first Kubernetes release where an initial version of the KEP was available
+- the version of Kubernetes where the KEP graduated to general availability
+- when the KEP was retired or superseded
+-->
+
+- 18.09.2023: KEP created (1.29)
+
+## Drawbacks
+
+<!--
+Why should this KEP _not_ be implemented?
+-->
+
+An estimated drawback is the change in user expectations.
+The users may expect that the `admin.conf` file will continue to have the
+super powers provided by the `system:masters` Group. This feature will affect
+this expectation. An action will be required by the same users to sign a new
+explicit `super-admin.conf` file with the super powers.
```

Comment:
Are some installers using this for rollback cluster?

Reply:
the new admin.conf that binds to the cluster-admin clusterrole, but not to the system:masters group can still be used for rollback. IMO.

the above talks about a case where users expect the admin.conf to be able to bypass RBAC, which will not be true once we create the separate super-admin.conf. the super-admin.conf will be able to bypass RBAC, but the admin.conf will not be able to. this can be covered in the release note.